cypress/e2e/middleware/enhanced.cy.ts

```diff
@@ -1,5 +1,24 @@
 describe('Enhanced middleware', () => {
-  it('rewrites the response body', () => {
+  it('rewrites the response body using request.rewrite()', () => {
+    cy.visit('/request-rewrite')
+    cy.get('#message').contains('This was static (& escaping test &amp;) but has been transformed in')
+    cy.contains("This is an ad that isn't shown by default")
+  })
+
+  it('modifies the page props when using request.rewrite()', () => {
+    cy.visit('/request-rewrite')
+    const data = cy.get('script#__NEXT_DATA__')
+    data.should('contain', '"message":"This was static (& escaping test &amp;) but has been transformed in')
```

Comment:
Consider testing that the `props.pageProps.message` property includes that text instead of just checking the raw string.

```suggestion
    const data = cy.get('script#__NEXT_DATA__').then((element) => {
      const { props } = JSON.parse(element.text());
      expect(props.pageProps.message).to.include('This was static (& escaping test &amp;) but has been transformed in')
    })
```